pkg/agent/cniserver/pod_configuration.go

```diff
@@ -666,3 +631,129 @@ func (pc *podConfigurator) reconcile(pods []corev1.Pod) error {
 	}
 	return nil
 }
+
+// connectInterfaces connect an existing interface to ovs br-int.
+func (pc *podConfigurator) connectInterfaceToOVS(
+	podName string,
+	podNameSpace string,
+	containerID string,
+	hostIface *current.Interface,
+	containerIface *current.Interface,
+	ips []*current.IPConfig,
+) error {
+	// Use the outer veth interface name as the OVS port name.
+	ovsPortName := hostIface.Name
+	containerConfig := buildContainerConfig(ovsPortName, containerID, podName, podNameSpace, containerIface, ips)
+
+	// create OVS Port and add attach container configuration into external_ids
+	klog.V(2).Infof("Adding OVS port %s for container %s", ovsPortName, containerID)
+	portUUID, err := pc.setupContainerOVSPort(containerConfig, ovsPortName)
+	if err != nil {
+		return fmt.Errorf("failed to add OVS port for container %s: %v", containerID, err)
+	}
+	// Remove OVS port if any failure occurs in later manipulation.
+	defer func() {
+		if err != nil {
+			_ = pc.ovsBridgeClient.DeletePort(portUUID)
+		}
+	}()
+
+	// GetOFPort will wait for up to 1 second for OVSDB to report the OFPort number.
+	ofPort, err := pc.ovsBridgeClient.GetOFPort(ovsPortName)
+	if err != nil {
+		return fmt.Errorf("failed to get of_port of OVS port %s: %v", ovsPortName, err)
+	}
+
+	klog.V(2).Infof("Setting up Openflow entries for container %s", containerID)
+	err = pc.ofClient.InstallPodFlows(ovsPortName, containerConfig.IP, containerConfig.MAC, pc.gatewayMAC, uint32(ofPort))
+	if err != nil {
+		return fmt.Errorf("failed to add Openflow entries for container %s: %v", containerID, err)
+	}
+	defer func() {
+		if err != nil {
+			_ = pc.ofClient.UninstallPodFlows(ovsPortName)
+		}
+	}()
+
+	containerConfig.OVSPortConfig = &interfacestore.OVSPortConfig{PortUUID: portUUID, OFPort: ofPort}
+	// Add containerConfig into local cache
+	pc.ifaceStore.AddInterface(containerConfig)
+	return nil
+}
+
+// disconnectInterfaceFromOVS disconnects an existing interface from ovs br-int.
+func (pc *podConfigurator) disconnectInterfaceFromOVS(containerConfig *interfacestore.InterfaceConfig) error {
+	containerID := containerConfig.ContainerID
+	klog.V(2).Infof("Deleting Openflow entries for container %s", containerID)
+	if err := pc.ofClient.UninstallPodFlows(containerConfig.InterfaceName); err != nil {
+		return fmt.Errorf("failed to delete Openflow entries for container %s: %v", containerID, err)
```

Comment:
we can ...
 this just extracts the existing implementation and places them under  disconnectInterfaceFromOVS, there is no change to error handling logic.

Reply:
I know. Do not mean your change. You can choose to ignore this comment, if you think it should be a different PR.

Reply:
Keep the comment a while to get @tnqn and @weiqiangt 's notice.

Reply:
@jianjuns If it failed to uninstall flow and succeeded to delete OVS port, the issue of reusing ofport could happen when container runtime calls CNI DEL next time. 
Since we expect the error should lead to CNI DEL retry, it doesn't matter that we don't remove the interface OVS port this round and bind lifecycle of flows and port. In worst case restarting agent can clean up stale ports.

Reply:
Ok. I guess we need a comment here.